android/app/src/main/java/app/organicmaps/util/Config.java

```diff
@@ -350,6 +333,57 @@ public static String getDonateUrl()
     return getString(KEY_DONATE_URL);
   }
 
+  public static void updateCounters(@NonNull Context context)
+  {
+    PreferenceManager.setDefaultValues(context, R.xml.prefs_main, false);
+
+    final SharedPreferences prefs = MwmApplication.prefs(context);
+    final SharedPreferences.Editor editor = prefs.edit();
+
+    editor.putLong(KEY_APP_LAUNCH_NUMBER, prefs.getLong(KEY_APP_LAUNCH_NUMBER, 0) + 1);
+
+    if (prefs.getInt(KEY_APP_LAST_INSTALL_VERSION, 0) != BuildConfig.VERSION_CODE ||
+        !prefs.getString(KEY_APP_LAST_INSTALL_FLAVOR, "").equals(BuildConfig.FLAVOR))
+    {
+      // A new version.
+      editor.putString(KEY_APP_LAST_INSTALL_FLAVOR, BuildConfig.FLAVOR);
+      editor.putInt(KEY_APP_LAST_INSTALL_VERSION, BuildConfig.VERSION_CODE);
+      editor.putInt(KEY_APP_SESSION_NUMBER, 0);
+      editor.putLong(KEY_APP_LAST_SESSION_TIMESTAMP, System.currentTimeMillis());
+      if (prefs.getInt(KEY_APP_FIRST_INSTALL_VERSION, 0) == 0)
+      {
+        editor.putString(KEY_APP_FIRST_INSTALL_FLAVOR, BuildConfig.FLAVOR);
+        editor.putInt(KEY_APP_FIRST_INSTALL_VERSION, BuildConfig.VERSION_CODE);
+      }
+    }
+    else
+    {
+      // An existing version.
+      long lastSessionTimestamp = prefs.getLong(KEY_APP_LAST_SESSION_TIMESTAMP, 0);
+      if (!DateUtils.isToday(lastSessionTimestamp))
```

Comment:
Always update KEY_APP_LAST_SESSION_TIMESTAMP ? Why this additional _today_ check?

Reply:
If the app was launched the first time today – count it as a new session... This is my guess.

Reply:
I have removed any tracking of sessions from the code.